pages/admin/lessons/[lessonId]/[pageName]/index.tsx

```diff
@@ -0,0 +1,185 @@
+import { gql, useQuery } from '@apollo/client'
+import { GetAppProps, withGetApp } from '../../../../../graphql'
+import { get, sortBy, toNumber, toUpper } from 'lodash'
+import React, { useEffect, useMemo, useState } from 'react'
+import AdminLessonNav from '../../../../../components/admin/lessons/AdminLessonNav'
+import AdminLessonSideNav from '../../../../../components/admin/lessons/AdminLessonSideNav'
+import AdminModuleInputs, {
+  Props
+} from '../../../../../components/admin/lessons/AdminModuleInputs'
+import Breadcrumbs from '../../../../../components/Breadcrumbs'
+import styles from '../../../../../scss/modules.module.scss'
+import navStyles from '../../../../../scss/adminLessonNav.module.scss'
+import { AdminLayout } from '../../../../../components/admin/AdminLayout'
+import { useRouter } from 'next/router'
+import Link from 'next/link'
+
+const MODULES = gql`
+  query {
+    modules {
+      id
+      name
+      content
+      lesson {
+        id
+      }
+      order
+    }
+  }
+`
+
+type Module = {
+  id: number
+  name: string
+  content: string
+  lesson: { id?: number }
+  order: number
+}
+type Modules = Module[]
+
+type ModulesTabProps = {
+  filteredModules: Modules
+  lessonId: number
+  refetch: Props['refetch']
+}
+
+const ModulesTab = ({
+  filteredModules,
+  lessonId,
+  refetch
+}: ModulesTabProps) => {
+  const [selectedIndex, setSelectedIndex] = useState(-1)
+  const onAddItem = () => setSelectedIndex(-1)
+  const onSelect = (item: Omit<Module, 'order'>) => setSelectedIndex(item.id)
+
+  useEffect(() => setSelectedIndex(get(filteredModules[0], 'id')), [lessonId])
+
+  return (
+    <div className={styles.container__modulesPanel}>
+      <AdminLessonSideNav
+        title="Modules"
+        items={filteredModules}
+        onAddItem={onAddItem}
+        onSelect={onSelect}
+        selectedIndex={selectedIndex}
+      />
+      <div className={styles.container__modulesPanel__inputs}>
+        <AdminModuleInputs
+          lessonId={lessonId}
+          refetch={refetch}
+          module={filteredModules.find(module => module.id === selectedIndex)}
+        />
+      </div>
+    </div>
+  )
+}
+
+const Lessons = ({ data }: GetAppProps) => {
+  const router = useRouter()
+  const { pageName, lessonId } = router.query
+
+  const { lessons } = data
+  const { data: modulesData, refetch } = useQuery<{ modules: Modules }>(MODULES)
+
+  const [lesson, setLesson] = useState({
```

Comment:
you don't need this as a state, because lessonId is coming from router

Reply:
Yes, but it's needed by the `Breadcrumbs` component

Reply:
I meant you could do this:

```js
const lesson = lessons.find(e => e.id === toNumber(lessonId)) || {title: '', id: -1}
```

You don't need a `lesson` state.